File: Level 2/콜라츠 추측/콜라츠 추측.py
def collatz(num):
    answer = 0
    while num != 1:
        if answer == 500:
            return -1
        elif num % 2 == 0:
            num /= 2
            answer += 1
        else:
            num = num * 3 + 1
            answer += 1
    return answer

# 아래는 테스트로 출력해 보기 위한 코드입니다.
print(collatz(6))

File: Level 2/콜라츠 추측/콜라츠 추측.cpp
#include<iostream>
using namespace std;

int collatz(int num)
{
	int answer = 0;
	while (num != 1) {
		if (answer == 500) {
			return -1;
		}
		else if (num % 2 == 0) {
			num /= 2;
			answer++;
		}
		else {
			num = num * 3 + 1;
			answer++;
		}
	}
	return answer;
}

int main()
{
	int testCase = 6;
	int testAnswer = collatz(testCase);

	cout << testAnswer;
}